addons/nametags/XEH_postInit.sqf

```diff
@@ -24,4 +26,6 @@ GVAR(ShowNamesTime) = -10;
 
 
 // Draw handle
-addMissionEventHandler ["Draw3D", {_this call FUNC(onDraw3d);}];
+if (GVAR(showPlayerNames) > 0 || GVAR(showVehicleCrewInfo)) then {
+    addMissionEventHandler ["Draw3D", {_this call FUNC(onDraw3d);}];
```

Comment:
If the GVAR(showPlayerNames) is initially zero and you turn it on later, they won't draw due to draw3d eh not being installed.

You need to monitor the "SettingsChanged" EH to check for that happening and add remove the Draw3D event accordingly. Check viewdistance for a possible implementation.